spinoso-string/src/lib.rs

```diff
@@ -1922,17 +1922,13 @@ impl String {
     /// use core::ops::Range;
     /// use spinoso_string::String;
     ///
-    /// let s = String::utf8(b"abc\xF0\x9F\x92\x8E\xFF".to_vec()); // "abc💎\xFF"
-    /// assert_eq!(s.get_char_slice(Range { start:  0, end:  1 }), Some(&b"a"[..]));
-    /// assert_eq!(s.get_char_slice(Range { start:  0, end:  3 }), Some(&b"abc"[..]));
-    /// assert_eq!(s.get_char_slice(Range { start:  0, end:  4 }), Some("abc💎".as_bytes()));
-    /// assert_eq!(s.get_char_slice(Range { start:  0, end:  5 }), Some(&b"abc\xF0\x9F\x92\x8E\xFF"[..]));
-    /// assert_eq!(s.get_char_slice(Range { start:  3, end: 10 }), Some(&b"\xF0\x9F\x92\x8E\xFF"[..]));
-    /// assert_eq!(s.get_char_slice(Range { start:  4, end: 10 }), Some(&b"\xFF"[..]));
+    /// let s = String::ascii(b"abc".to_vec());
+    /// assert_eq!(s.get_char_slice(Range { start:  1, end:  3 }), Some("bc".as_bytes()));
```

Comment:
Thanks for simplifying these tests. Now that I'm taking another peek here, I think we can also make an opportunistic refactor to replace the range structs with range literals. Something like this:

```suggestion
    /// assert_eq!(s.get_char_slice(1..3), Some("bc".as_bytes()));
```

Reply:
Fixed [here](https://github.com/artichoke/artichoke/pull/2102/files?diff=split&w=0#diff-a2d2fa818d5b61fb277dd19f5a3a58b696944224b071fe8faee5bca3c19103d3R1925-R1930)

Thanks for the suggestion! I also updated the [unit tests](https://github.com/artichoke/artichoke/pull/2102/files?diff=split&w=0#diff-edc577dd37d438dce282a0d8476d38f20897d1052364a5e8ef495bf09aa69852R1197-R1204） to use Range literals.

Reply:
> I also updated the [unit tests](https://github.com/artichoke/artichoke/pull/2102/files?diff=split&w=0#diff-edc577dd37d438dce282a0d8476d38f20897d1052364a5e8ef495bf09aa69852R1197-R1204) to use Range literals.

Opps I realized Clippy will [complain](https://github.com/artichoke/artichoke/runs/8207578734?check_suite_focus=true) for [reversed_empty_ranges](https://rust-lang.github.io/rust-clippy/master/index.html#reversed_empty_ranges) when I use `3..1` instead of `Range { start: 3, end: 1 }`. I added [`allow(clippy::reversed_empty_ranges)`](https://github.com/artichoke/artichoke/pull/2102/commits/229c2454034463cc7cc438037aea04bbec9c173e#diff-1578dd864d1a9bac39c945d985837664c4c4655d41f16fdba68ea16ecef49c4aR535) for now for all three `get_char_slice_invalid_range` unit tests, but please let me know if reverting back to using `Range` is preferred!